nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -93,136 +68,53 @@ private static Random getRandomnessSource(String source) {
-            } finally {
-                ControlNondeterminism.shouldOutputTrace = true;
+        if (shouldFlip) {
+            for (int i = 0; i < originalArrays.length; i++) {
+                originalArrays[i] = Arrays.copyOf(originalArrays[i], originalArrays[i].length + 1);
             }
         }
-    }
 
-    private static boolean isDebuggingUniquePoint() {
-        return ControlNondeterminism.config.start >= 0 && ControlNondeterminism.config.end >= 0
-                && ControlNondeterminism.config.start == ControlNondeterminism.config.end
-                && ControlNondeterminism.count == ControlNondeterminism.config.start;
-    }
+        return originalArrays;
 
-    private static boolean shouldExploreForInstance() {
-        return ControlNondeterminism.count >= ControlNondeterminism.config.start
-                && ControlNondeterminism.count <= ControlNondeterminism.config.end;
     }
 
-    private static String getSource() {
-        return Thread.currentThread().getStackTrace()[3].toString();
-
-    }
 
     private static class JVMShutdownHook extends Thread {
         @Override
         public void run() {
-            ControlNondeterminism.config.createNondexDirIfNeeded();
+            nondex.getConfig().createNondexDirIfNeeded();
             try {
-                int localCount = ControlNondeterminism.count;
-                int localShufflesCount = ControlNondeterminism.shuffleCount;
-                Files.write(ControlNondeterminism.config.getConfigPath(),
-                        ControlNondeterminism.config.toString().getBytes(),
-                        StandardOpenOption.CREATE,
+                int localCount = nondex.getPossibleExplorations();
+                int localShufflesCount = nondex.getActualExplorations();
+                Files.write(nondex.getConfig().getConfigPath(),
+                        nondex.getConfig().toString().getBytes(), StandardOpenOption.CREATE,
                         StandardOpenOption.APPEND);
-                Files.write(ControlNondeterminism.config.getInvocationsPath(),
+                Files.write(nondex.getConfig().getInvocationsPath(),
                         ("COUNT:" + localCount + String.format("%n")).getBytes(),
                         StandardOpenOption.CREATE, StandardOpenOption.APPEND);
-                Files.write(ControlNondeterminism.config.getInvocationsPath(),
+                Files.write(nondex.getConfig().getInvocationsPath(),
                         ("SHUFFLES:" + localShufflesCount + String.format("%n")).getBytes(),
                         StandardOpenOption.APPEND);
             } catch (IOException ioe) {
-                Logger.getGlobal().log(Level.SEVERE, "Exception when printing shuflling counts in shutdown hook.", ioe);
+                Logger.getGlobal().log(Level.SEVERE,
+                        "IOException when printing shuffling counts in shutdown hook.", ioe);
+            } catch (Throwable ex) {
```

Comment:
Codacy keeps flagging this (and sending emails...), is there some Throwable that can be thrown here that Exception would not catch?